### PR TITLE
add username suggestions functionality

### PR DESCRIPTION
**Pull request by waheedahmed**
Added username suggestions functionality according to redesign.

Backend PR: https://github.com/edx/edx-platform/pull/27387

https://user-images.githubusercontent.com/2851134/115433041-82ad9800-a220-11eb-92b7-f9eb8d6ae4d2.mp4

VAN-52

### CODE REVIEW

File: src/register/RegistrationPage.jsx [hunk 1]
```diff
@@ -229,6 +230,21 @@ class RegistrationPage extends React.Component {
...
+    const payload = {
+      username: suggestion,
+    };
+    this.setState({
+      ...payload,
+      fieldName,
+      updateFieldErrors: false,
+      updateAlertErrors: false,
+    }, () => {
+      this.validateInput(fieldName, suggestion, payload, false);
```
Comment (zainab-amir) [inline src/register/RegistrationPage.jsx:244]: why do we need to validate the suggestion? is it not already verified before backend sends it to the frontend?

Comment (waheedahmed) [reply src/register/RegistrationPage.jsx:244]: @zainab-amir It's just to clear out the error message and username suggestions.

File: src/register/UsernameField.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { usernameSuggestionsSelector } from './data/selectors';
+import { FormGroup } from '../common-components';
+import messages from './messages';
+
+const UsernameField = (props) => {
+  const { intl, usernameSuggestions } = props;
+
+  return (
+    <FormGroup {...props}>
+      {usernameSuggestions && usernameSuggestions.length > 0 ? (
```
Comment (zainab-amir) [inline src/register/UsernameField.jsx:17]: if the default value is `[]` do we need to add the first check, shouldn't `usernameSuggestions.length` be enough?

**Review by zainab-amir — APPROVED** [review]

File: src/register/UsernameField.jsx (hunk 2 above)
Comment (waheedahmed) [reply src/register/UsernameField.jsx:17]: Yes, it should be.

**Pull request merged**